Px.Utils/PxFile/PxFileMetadataReader.cs

```diff
@@ -234,6 +234,27 @@ public static Encoding GetEncoding(Stream stream, PxFileSyntaxConf? symbolsConf
             return GetEncodingFromValue(encoding.Value, symbolsConf);
         }
 
+        /// <summary>
+        /// Tries to determine the encoding of the provided stream based on the Byte Order Mark (BOM) or the CODEPAGE keyword in the metadata.
+        /// </summary>
+        /// <param name="stream">The stream from which to determine the encoding.</param>
+        /// <param name="encoding">Encoding format the function tries to return</param>
+        /// <param name="symbolsConf">The symbols configuration to use when reading the metadata. If not specified the default configuration is used.</param>
+        /// <returns></returns>
+        public static bool TryGetEncoding(Stream stream, out Encoding? encoding, PxFileSyntaxConf? symbolsConf = null)
+        {
+            try
```

Comment:
Dont throw exceptions inside Try functions. The idea of these type of functions is that they enable failing the thing they are doing without exceptions, and the user will most likely expect them to behave that way.

Reply:
Fixed